lib/rubygems/package.rb

```diff
@@ -371,6 +375,8 @@ def extract_tar_gz io, destination_dir, pattern = "*" # :nodoc:
           FileUtils.chmod entry.header.mode, destination
         end if entry.file?
 
+        File.symlink(entry.header.linkname, destination) if entry.symlink?
+
```

Comment:
This chunk is a little more involved.  Like the [`extract_files_absolute`](https://github.com/rubygems/rubygems/blob/master/test/rubygems/test_gem_package.rb#L384) test a gem must not be installable if it symlinks outside of the gem's install directory.